### PR TITLE
fix(frontend, public-profile): File size limit upgraded to 10MB

### PR DESCRIPTION
**Pull request by grignards**
See #4101

And use blobs for previewing the image in the browser.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link): testing the limit would imply to have a big image in the repository (or generate it when testing).
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

**Testing method**

Try to upload an image >1MB but <10MB, it should be possible. With an image >10MB the popup should refuse to upload the file.
Check that the preview of the image is properly displayed.

### CODE REVIEW

File: frontend_lib/src/container/PopupUploadFile.jsx [hunk 1]
```diff
@@ -138,20 +138,14 @@ class PopupUploadFile extends React.Component {
...
-      reader.onload = e => {
-        const img = new Image()
-        img.src = e.target.result
-        img.onerror = () => reject(new Error())
-        img.onload = () => {
-          if (e.total > 0) {
-            resolve(e.target.result)
-          } else reject(new Error())
-        }
+      img.onerror = () => reject(new Error())
```
Comment (raphj) [inline frontend_lib/src/container/PopupUploadFile.jsx:145]:
```suggestion
      img.onerror = () => reject(new Error('Failed to load the image'))
```

**Review by gbonaspetti — APPROVED** [review]

**Comment by PhilippeAccorsi** [comment]
Is it not possible to use paramter like this in development.ini to define the max size of file ?
`limitation.content_length_file_size`

**Comment by grignards** [comment]
> Is it not possible to use paramter like this in development.ini to define the max size of file ?
> `limitation.content_length_file_size`

This parameter has no effect on avatars & covers, but it could be a good idea to take it into account.

**Comment by gbonaspetti** [comment]
Tested on branch, but the avatar takes a long time to change, maybe put a loader during that time.

**Comment by grignards** [comment]
> Tested on branch, but the avatar takes a long time to change, maybe put a loader during that time.

Adding a loader during the load of an image seems to be tricky so I'd rather postpone it. An alternative way would be to change the default compression level used in preview generator.

**Comment by gbonaspetti** [comment]
> > Tested on branch, but the avatar takes a long time to change, maybe put a loader during that time.
> 
> Adding a loader during the load of an image seems to be tricky so I'd rather postpone it. An alternative way would be to change the default compression level used in preview generator.

And the alternative way is quick to develop?

**Comment by gbonaspetti** [comment]
> > And the alternative way is quick to develop?
> 
> Yes I think it is quick to develop, but implies a change in preview-generator + its release + a change in the requirements of the backend.

@grignards I think you edited my answer :smile: I'm gonna approved the test part like that, but maybe it's cool to make an issue about it.

**Comment by gbonaspetti** [comment]
Tested on branch.

**Comment by grignards** [comment]
> @grignards I think you edited my answer smile I'm gonna approved the test part like that, but maybe it's cool to make an issue about it.

Issue in preview generator: https://github.com/algoo/preview-generator/issues/211

**Pull request merged**